api/envoy/config/route/v3/route_components.proto

```diff
@@ -1783,6 +1783,9 @@ message RateLimit {
     message RemoteAddress {
       option (udpa.annotations.versioning).previous_message_type =
           "envoy.api.v2.route.RateLimit.Action.RemoteAddress";
+
+      // Length of prefix, e.g. 0, 32. Defaults to -1 when unset.
+      google.protobuf.UInt32Value prefix_len = 1 [(validate.rules).uint32 = {lte: 128}];
```

Comment:
A few points on the API:

1) What does "-1" mean in the context of an unsigned int? I'm not sure what this will evaluate to.
2) More generally, I think we have to explicitly handle v4 and v6 here. I would probably do something like have a v4_prefix_len and a v6_prefix_len, where the default if unset is /128 and /32 respectively?
3) What does "length of prefix" actually mean in practice? I assume it will just mask before sending? Can you add some examples to the API docs?

Reply:
you're right, `-1` is a bad design.
`prefix_len` comes from [here](https://github.com/envoyproxy/envoy/blob/9d5627a0879b0a029e90515137c108e1d2884bfc/api/envoy/api/v2/core/address.proto#L134), but `0` is not good as default value `In case of the prefix_len == 0, then it will get an any-address`.

`v4_prefix_len ` and `v6_prefix_len` is a good idea, I'll update soon.

Reply:
But can you clarify what this actually does? Does it mask before sending? If so can we call it something like v4_prefix_mask_len?

Reply:
I would like to also ask for some practical guidelines for how this would be used in practice. For example, is this only intened to be used in RFC1918 environments where subnet sizes are well understood? If it's for use on public IP, how do you pick these sizes?

Reply:
> But can you clarify what this actually does? Does it mask before sending? If so can we call it something like v4_prefix_mask_len?

we deploy istio with Kubernetes, envoy run as sidecar container, [here](https://istio.io/latest/docs/tasks/policy-enforcement/rate-limit/#local-rate-limit) is the doc on istio.io site.

service topology:

![sidecar](https://user-images.githubusercontent.com/4354057/164351200-a568ca1f-4395-4024-81e5-ad808a6aa6d2.PNG)

right now, I can apply ratelimit rules(sleep 10/min, fortio 100/min) base on remote address(mostly podip in my case), with a custom RLS using GlobalRateLimit.
we want use this with LocalRateLimit, currently we need re-apply the `EnvoyFilter` once `fortio` or `sleep`'s podip changes.

Reply:
> If it's for use on public IP, how do you pick these sizes?

I may miss something, I do not think too much about the difference between public IPs and private IPs.

Reply:
I still don't understand what this does from the description. The only thing that makes sense to me is masking. Can you update the docs with specific examples? Thank you.

Reply:
Assuming masking as Matt points out, IMHO, the only way this works sensibly is if you have a networking environment in which traffic comes from IP addresses that are known to live in subnets of a given size. If you were operating a public Internet facing ingress gateway, it seems like this would be a pretty useless. But if this is some network you control and you know k8s pods come from well-known subnet ranges, this is just a way to ratelimit on on subnet. So, I think you need to document clearly the places this feature should be used in the Envoy docs to make this crisp.